backend/pom.go

```diff
@@ -0,0 +1,129 @@
+)
+
+const (
+	// PomFilename is the file name used by the pomfiles.
+	PomFilename = "pom.xml"
+)
+
+// Pom is a backend for Pom or Project Object Model files. It is an xml file
+// commonly used by the Maven Project under the name pom.xml. We are getting
+// license name by parsing the pom.xml file.
+type Pom struct {
+	Debug bool
+	Logf  func(format string, v ...interface{})
+}
+
+// String method returns the name of the backend.
+func (obj *Pom) String() string {
+	return "pom"
+}
+
+// ScanData method is used to extract license ids from data and return licenses
+// based on the license ids.
+func (obj *Pom) ScanData(ctx context.Context, data []byte, info *interfaces.Info) (*interfaces.Result, error) {
+	// this check is taking place with the assumption that the file that will be
+	// scanned will have to be named "pom.xml"
+	if info.FileInfo.Name() != PomFilename {
+		return nil, nil // skip
+	}
+	if info.FileInfo.IsDir() {
+		return nil, nil // skip
+	}
+	if len(data) == 0 {
+		return nil, nil // skip
+	}
+
+	licenseMap := make(map[string]struct{})
+	var pomFileLicenses PomLicenses
+
+	// parsing pom.xml file to get license names in struct
+	if err := xml.Unmarshal(data, &pomFileLicenses); len(pomFileLicenses.Names) == 0{
+		// NOTE: This is NOT the same as as interfaces.ErrUnknownLicense since here we
+		// are checking if we managed to get any licenses regardless of their
+		// validity. If we did not get any licenses we return nil and any error
+		// that took place while parsing.
+		return nil, errwrap.Wrapf(err, "parse error")
+	}
+	// lid is license id
+	for _, lid := range pomFileLicenses.Names {
+		licenseMap[lid] = struct{}{}
+	}
+
+	ids := []string{}
+	for id := range licenseMap {
+		ids = append(ids, id)
+	}
+	sort.Strings(ids) // deterministic order
+
+	licenseList := []*licenses.License{}
+
+	for _, id := range ids {
+		license := &licenses.License{
+			SPDX: id,
+			// TODO: populate other fields here?
+		}
+
+		// If we find an unknown SPDX ID, we don't want to error, because that would
+		// allow someone to put junk in their code to prevent us scanning it. Instead,
+		// create an invalid license but return it anyways. If we ever want to check
+		// validity, we know to expect failures.
+		// XXX: Many Pom licenses are not SPDX, therefore we might want to add an alias
+		// matcher in the future
+		if err := license.Validate(); err != nil {
+			//return nil, err
+			license = &licenses.License{
+				//SPDX: "",
+				Origin: "", // unknown!
+				Custom: id,
+				// TODO: populate other fields here (eg: found license text)
+			}
+		}
+
+		licenseList = append(licenseList, license)
+	}
+
+	result := &interfaces.Result{
+		Licenses:   licenseList,
+		Confidence: 1.0, // TODO: what should we put here?
+	}
+
+	return result, nil
+}
+
+// PomLicenses is a struct that helps store license names from the licenses
+// field in a pom.xml file.
+type PomLicenses struct {
+	Names []string `xml:"licenses>license>name"`
```

Comment:
docstring

Reply:
Fixed, added docstring for Names variable.